rllib/tests/test_execution.py

```diff
@@ -204,6 +211,38 @@ def test_straggler_rollouts(ray_start_regular_shared):
     workers.stop()
 
 
+def test_all_straggler_rollouts(ray_start_regular_shared):
+    slow_factor = 7
+    workers = make_slow_sim_workers(2, lambda i: slow_factor if i > 0 else False)
+    a = ParallelRollouts(workers, mode="bulk_sync")
+    batch = None
+    tries = 0
+    while not batch:
+        tries += 1
+        try:
+            batch = next(a)
+        except StopIteration:
+            continue
+    assert batch.count == 200
+    counters = a.shared_metrics.get().counters
+    assert counters["num_steps_sampled"] == 200, counters
+    assert tries > 600 and tries < 700
+    a = ParallelRollouts(workers, mode="async")
+    batch = None
+    tries = 0
+    while not batch:
+        tries += 1
+        try:
+            batch = next(a)
+        except StopIteration:
+            continue
+    assert batch.count == 100
```

Comment:
Comments explaining these test cases would help. I think I can intuit what's being established here, but it's not obvious.